librz/analysis/types.c

```diff
@@ -0,0 +1,85 @@
+// SPDX-FileCopyrightText: 2021-2023 Anton Kochkov <anton.kochkov@gmail.com>
+// SPDX-License-Identifier: LGPL-3.0-only
+
+#include <rz_util.h>
+#include <rz_type.h>
+#include <rz_analysis.h>
+#include <string.h>
+
+static RZ_OWN RzList /* <RzTypePathTuple*> */ *var_global_type_paths(RzAnalysis *analysis, RzAnalysisVarGlobal *gv, ut64 addr) {
+	rz_return_val_if_fail(gv, false);
+
+	RzType *t = gv->type;
+
+	// Handle only identifiers here
+	if (t->kind != RZ_TYPE_KIND_IDENTIFIER) {
```

Comment:
Only as a hint, there is also `rz_type_path_by_offset()` which would let you skip a lot of this inlined logic and supports even more kinds of types like arrays. But I see this code was only moved here as-is so changes like that should go in another commit then.

Reply:
Yes, my plan is to complete migration first, then there are a few aspects of what could be improved: 1) display 2) propagation 3) integration with DWARF/PDB/whatever. The plan is to do it after this PR is merged though, but before 0.6.0

Reply:
Agreed this is the way to go.

Reply:
While solving side effect cancer, I did this anyway, to simplify the code.